### ISSUE TITLE
Unexpected CryptoAPI failure generating seed

### ISSUE THREAD

**Issue by lemaxw** [nats-io/nats.java#328]
Hi,

Sometimes (rare) get on connect following error stack:
 2020-07-20 08:11:39,056 [main] DEBUG [] - java.lang.InternalError: Unexpected CryptoAPI failure generating seed
....
Caused by: java.lang.InternalError: Unexpected CryptoAPI failure generating seed
	at java.base/sun.security.provider.NativeSeedGenerator.getSeedBytes(NativeSeedGenerator.java:62)
	at java.base/sun.security.provider.SeedGenerator.generateSeed(SeedGenerator.java:144)
	at java.base/sun.security.provider.SecureRandom.engineGenerateSeed(SecureRandom.java:139)
	at java.base/java.security.SecureRandom.generateSeed(SecureRandom.java:847)
	at io.nats.client.NUID.<init>(NUID.java:77)
	at io.nats.client.impl.NatsConnection.<init>(NatsConnection.java:157)
	at io.nats.client.impl.NatsImpl.createConnection(NatsImpl.java:28)
	at io.nats.client.Nats.createConnection(Nats.java:247)
	at io.nats.client.Nats.connect(Nats.java:164)

Here is log of natsserver (at 08:11:36, natsserver was restarted due to above exception):
[4612] 2020/07/20 08:09:34.805186 [DBG] 127.0.0.1:52578 - cid:23 - Starting TLS client connection handshake
[4612] 2020/07/20 08:09:34.816185 [DBG] 127.0.0.1:52578 - cid:23 - TLS handshake complete
[4612] 2020/07/20 08:09:34.816185 [DBG] 127.0.0.1:52578 - cid:23 - TLS version 1.2, cipher suite TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
[4612] 2020/07/20 08:09:34.817189 [TRC] 127.0.0.1:52578 - cid:23 - <<- [CONNECT {"lang":"java","version":"2.6.6","protocol":1,"verbose":false,"pedantic":false,"tls_required":true,"echo":true}]
[4612] 2020/07/20 08:09:34.817189 [TRC] 127.0.0.1:52578 - cid:23 - <<- [PING]
[4612] 2020/07/20 08:09:34.817189 [TRC] 127.0.0.1:52578 - cid:23 - ->> [PONG]
[4612] 2020/07/20 08:09:34.818184 [DBG] 127.0.0.1:52578 - cid:23 - Client connection closed
[4612] 2020/07/20 08:09:40.703923 [TRC] 10.0.0.13:52498 - cid:16 - <<- [PING]
[4612] 2020/07/20 08:09:40.703923 [TRC] 10.0.0.13:52498 - cid:16 - ->> [PONG]
[4612] 2020/07/20 08:09:42.744837 [DBG] 10.0.0.13:52498 - cid:16 - Client Ping Timer
[4612] 2020/07/20 08:09:42.744837 [DBG] 10.0.0.13:52498 - cid:16 - Delaying PING due to remote ping 2s ago
[4612] 2020/07/20 08:10:03.143813 [TRC] 10.0.0.13:52433 - cid:6 - <<- [PING]
[4612] 2020/07/20 08:10:03.143813 [TRC] 10.0.0.13:52433 - cid:6 - ->> [PONG]
[4612] 2020/07/20 08:10:05.492701 [DBG] 10.0.0.13:52433 - cid:6 - Client Ping Timer
[4612] 2020/07/20 08:10:05.492701 [DBG] 10.0.0.13:52433 - cid:6 - Delaying PING due to remote ping 2s ago
[4612] 2020/07/20 08:10:34.825623 [DBG] 127.0.0.1:60691 - cid:24 - Client connection created
[4612] 2020/07/20 08:10:34.825623 [DBG] 127.0.0.1:60691 - cid:24 - Starting TLS client connection handshake
[4612] 2020/07/20 08:10:34.836626 [DBG] 127.0.0.1:60691 - cid:24 - TLS handshake complete
[4612] 2020/07/20 08:10:34.836626 [DBG] 127.0.0.1:60691 - cid:24 - TLS version 1.2, cipher suite TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
[4612] 2020/07/20 08:10:34.838626 [TRC] 127.0.0.1:60691 - cid:24 - <<- [CONNECT {"lang":"java","version":"2.6.6","protocol":1,"verbose":false,"pedantic":false,"tls_required":true,"echo":true}]
[4612] 2020/07/20 08:10:34.838626 [TRC] 127.0.0.1:60691 - cid:24 - <<- [PING]
[4612] 2020/07/20 08:10:34.838626 [TRC] 127.0.0.1:60691 - cid:24 - ->> [PONG]
[4612] 2020/07/20 08:10:34.839625 [DBG] 127.0.0.1:60691 - cid:24 - Client connection closed
[4896] 2020/07/20 08:11:36.924327 [INF] Starting nats-server version 2.1.4
[4896] 2020/07/20 08:11:36.935327 [DBG] Go build version go1.13.7
[4896] 2020/07/20 08:11:36.935327 [INF] Git commit [fb009af]
[4896] 2020/07/20 08:11:36.935327 [INF] Listening for client connections on 0.0.0.0:8480
[4896] 2020/07/20 08:11:36.936326 [INF] TLS required for client connections
[4896] 2020/07/20 08:11:36.936326 [INF] Server id is NAZZBOSLPUGLL22XCPFA2TOGLPIHQUJP35I5IWAZSZLRJNG33BIPIOMO
[4896] 2020/07/20 08:11:36.936326 [INF] Server is ready
[4896] 2020/07/20 08:11:36.936326 [DBG] Get non local IPs for "0.0.0.0"
[4896] 2020/07/20 08:11:36.938327 [DBG]  ip=10.0.0.13

**Comment by sasbury** [comment]
What platform and versions of things?

**Comment by lemaxw** [comment]
Hi,

It's windows 2012 server, java 11, nats 2.1.4, jnats 2.6.6.

Thanks!

**Comment by lemaxw** [comment]
Hi, 
Can you please comment? The issue reoccurs and happens even with closed tsl...

**Comment by scottf** [comment]
@lemaxw This should be fixed in the 2.9.x version (latest master branch). We replaced instantiation of SecureRandom where we specified the algorithm to use with the default construction. Please try the latest and let us know.

**Comment by lemaxw** [comment]
Hi, 
Great news!
@scottf , when do you plan release?
Thank you!

**Comment by scottf** [comment]
@lemaxw Don't hold me to it, but I think 2.9.0 is slated to release at the end of the month. You can use the source master branch if you like or can use the snapshot.

Maven (I think this is right)
```
<repositories>
    <repository>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>

<dependency>
    <groupId>io.nats</groupId>
    <artifactId>jnats</artifactId>
    <version>2.9.0-SNAPSHOT</version>
</dependency>
```

Gradle
```
repositories {
    ...
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }

dependencies {
    compile 'io.nats:jnats:2.9.0-SNAPSHOT'
```

**Comment by scottf** [comment]
Closed as fixed in the 2.10 release
